BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python3
+"""
+The current tooling supported in CMake, Homebrew, and QT5 are inufficient for
+creating MacOSX universal binaries automatically for applications like Dolphin
+which have more complicated build requirements (like different libraries, build
+flags and source files for each target architecture).
+
+So instead, this script manages the configuration and compilation of distinct
+builds and project files for each target architecture and then merges the two
+binaries into a single universal binary.
+
+Running this script will:
+1) Generate Xcode project files for the ARM build (if project files don't
+   already exist)
+2) Generate Xcode project files for the x64 build (if project files don't
+   already exist)
+3) Build the ARM project for the selected build_target
+4) Build the x64 project for the selected build_target
+5) Generates universal .app packages combining the ARM and x64 packages
+6) Utilizes the lipo tool to combine the binary objects inside each of the
+   packages into universal binaries
+7) Code signs the final universal binaries using the specified
+   codesign_identity
+"""
+
+import argparse
+import filecmp
+import glob
+import json
+import multiprocessing
+import os
+import shutil
+import subprocess
+
+# The config variables listed below are the defaults, but they can be
+# overridden by command line arguments see parse_args(), or run:
+# BuildMacOSUniversalBinary.py --help
+DEFAULT_CONFIG = {
+
+    # Location of destination universal binary
+    "dst_app": "universal/",
+    # Build Target (dolphin-emu to just build the emulator and skip the tests)
+    "build_target": "ALL_BUILD",
+
+    # Location for CMake to search for files(default is for homebrew
+    "arm64_cmake_prefix":  '/opt/homebrew',
+    "x86_64_cmake_prefix": '/usr/local',
+
+    # Locations to qt5 directories for arm and x64 libraries
+    # The default values of these paths are taken from the default
+    # paths used for homebrew
+    "arm64_qt5_path":  '/opt/homebrew/opt/qt5',
+    "x86_64_qt5_path": '/usr/local/opt/qt5',
+
+    # Identity to use for code signing. "-" indicates that the app will not
+    # be cryptographically signed/notarized but will instead just use a
+    # SHA checksum to verify the integrity of the app. This doesn't
+    # protect against malicious actors, but it does protect against
+    # running corrupted binaries and allows for access to the extended
+    # permisions needed for ARM builds
+    "codesign_identity":  '-',
+    # Entitlements file to use for code signing
+    "entitlements": "../Source/Core/DolphinQt/DolphinEmu.entitlements",
+
+    # Minimum macOS version for each architecture slice
+    "arm64_mac_os_deployment_target":  "11.0.0",
+    "x86_64_mac_os_deployment_target": "10.12.0",
+
+    # CMake Generator to use for building
+    "generator": "Unix Makefiles",
```

Comment:
Is there any reason we're not using Ninja?

Reply:
It isn't installed on macOS by default.
However, the buildbot uses Ninja.